lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_action.rb

```diff
@@ -228,32 +235,44 @@ def self.upload_binary(app_id, binary_path)
         connection.post("/app-binary-uploads?app_id=#{app_id}", File.open(binary_path).read) do |request|
           request.headers["Authorization"] = "Bearer " + auth_token
         end
+      rescue Faraday::ResourceNotFound
+        UI.crash!(ErrorMessage::INVALID_APP_ID)
```

Comment:
Similar suggestions to before, if you can give the inputs that _caused_ the error in the error message, the customer might have an easier time self-resolving.